spec/rubocop/cop/rspec/factory_bot/create_list_spec.rb

```diff
@@ -142,6 +142,39 @@
         end
       RUBY
     end
+
+    it 'flags usage of n.times with dynamic arguments' do
+      expect_offense(<<~RUBY)
+        3.times { create(:user, foo: rand(2), bar: 'baz', joe: doe) }
+        ^^^^^^^ Prefer create_list.
+      RUBY
+
+      expect_correction(<<~RUBY)
+create_list(:user, 3, bar: 'baz') do |user|
```

Comment:
The behaviour is surprisingly different, see https://github.com/thoughtbot/factory_bot/issues/1444

>  The model is saved when it’s created and then it’s passed to the block. If you want to update the model in the block, call `save` in the block.

So even though it resolves one issue with `rand` not being random across the list of records, but it leaves dirty attributes, which are not persisted.